src/hotspot/cpu/x86/templateTable_x86.cpp

```diff
@@ -3570,7 +3641,7 @@ void TemplateTable::prepare_invoke(int byte_no,
   load_invoke_cp_cache_entry(byte_no, method, index, flags, is_invokevirtual, false, is_invokedynamic);
 
   // maybe push appendix to arguments (just before return address)
-  if (is_invokedynamic || is_invokehandle) {
+  if (is_invokehandle) {
```

Comment:
Should the local variable `is_invokedynamic` be removed?

Reply:
I believe that local variable is now totally unused, but this is also true of `is_invokevfinal`. Since that local variable still exists despite being unused, I left `is_invokedynamic` as well. I think in the long term both of these variables should be removed.

Reply:
There's no `is_invokevfinal` in this scope (`prepare_invoke()`). `false` is passed as actual parameter `is_invokevfinal` to `load_invoke_cp_cache_entry()`. I thought the same could be done for the parameter `is_invokedynamic`.
Just commenting :)